trlx/model/accelerate_base_model.py

```diff
@@ -81,8 +88,7 @@ def __init__(self, config, train_mode=True):
                     }
                 },
             )
-
-        self.opt = torch.optim.AdamW(
+        self.opt = bnb.optim.AdamW8bit(
```

Comment:
Nice! 

This removes support for vanilla AdamW and some user devices may not be optimized for 8bit (requires Turing or Ampere GPUs). We should introduce an optimizer selection interface in the future but for now, maybe we could add a constructor flag to toggle between 8-bit and vanilla AdamW? What do you think?

Reply:
That'd probably be best, my only other idea is automatically detecting if bnb is installed but problems arise in bnb compiling without cuda support, and just that giving more control to the user is probably best